catalog/files/impl/src/main/java/org/projectnessie/catalog/files/s3/S3Clients.java

```diff
@@ -118,31 +115,12 @@ public static SdkHttpClient apacheHttpClient(S3Config s3Config, SecretsProvider
     return httpClient.buildWithDefaults(options.build());
   }
 
-  public static AwsCredentialsProvider basicCredentialsProvider(
-      Optional<String> accessKeyIdRef, Optional<String> secretAccessKeyIdRef) {
-    String key =
-        accessKeyIdRef.orElseThrow(
-            () -> new IllegalStateException("Secret reference to S3 access key ID is not defined"));
-
-    String secret =
-        secretAccessKeyIdRef.orElseThrow(
-            () ->
-                new IllegalStateException(
-                    "Secret reference to S3 secret access key is not defined"));
-
-    return () -> AwsBasicCredentials.create(key, secret);
-  }
-
   public static AwsCredentialsProvider awsCredentialsProvider(
       S3BucketOptions bucketOptions, S3Sessions sessions) {
-    Optional<String> role = bucketOptions.assumeRole();
-    if (role.isEmpty()) {
-      return basicCredentialsProvider(
-          bucketOptions.accessKey().map(BasicCredentials::name),
-          bucketOptions.accessKey().map(BasicCredentials::secret));
+    if (bucketOptions.assumeRole().isPresent()) {
```

Comment:
@dimas-b is it possible to define a role to be assumed, but keep using `REQUEST_SIGNING` for clients? I suppose in this case credentials would be fetched and cached, and the server would use them to sign requests?

Asking because the Helm chart currently assumes that if a role is provided, then `ASSUME_ROLE` is used for clients. Maybe we need to change that?